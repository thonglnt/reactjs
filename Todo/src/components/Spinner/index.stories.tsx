// Lib
import { Story } from '@storybook/react';

// Components
import Spinner from '@components/Spinner';

export default {
  title: 'Spinner',
  component: Spinner,
};

export const SpinnerSample: Story = () => <Spinner />; 