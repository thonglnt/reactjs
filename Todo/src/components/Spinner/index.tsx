export type SpinnerProps = {
  className?: string;
};

export const Spinner = ({ className }: SpinnerProps) => {
  return <div className={`loader ${className}`}></div>;
};

export default Spinner;
