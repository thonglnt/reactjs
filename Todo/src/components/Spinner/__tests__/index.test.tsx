// Lib
import { render } from '@testing-library/react';

// Components
import Spinner, { SpinnerProps } from '@components/Spinner';

describe('Test Spinner component', () => {
  test('It should render correctly with initial Props', () => {
    const { container } = render(<Spinner />);

    expect(container).toMatchSnapshot();
  });
});
