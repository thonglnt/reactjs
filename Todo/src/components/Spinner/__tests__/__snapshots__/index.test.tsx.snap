// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Spinner component It should render correctly with initial Props 1`] = `
<div>
  <div
    class="loader undefined"
  />
</div>
`;
